doc/features/metrics/README.md

```diff
@@ -0,0 +1,119 @@
+
+## Structure
+
+There are two categories of metrics:
+
+- **session-level**: the measured data is global to a `Session` instance. For example, `connected-nodes` measures the number of nodes to which we have connections.
+- **node-level**: the data is specific to a node (and therefore there is one metric instance per node). For example, `pool.open-connections` measures the number of connections open to this particular node.
+
+**Metric names** are path-like, dot-separated strings. An example is `errors.connection.init`. Metrics are represented by the classes `SessionMetric` and `NodeMetric`. These classes have `public static readonly` properties with all individual metrics that can be enabled on the driver. For instance, `NodeMetric.Counters.ConnectionInitErrors` is a `NodeMetric` object with a `Name` property that returns `errors.connection.init`.
+
+Metrics are grouped into **buckets** and each bucket has a name. The most common scenario is to concatenate the bucket name with the metric name which is what the [App.Metrics provider] does. The bucket name is the name of the session (see `Builder.WithSessionName()`) and, in the case of node-level metrics, nodes followed by a textual representation of the node’s address. If the client application specifies a bucket prefix with `DriverMetricsOptions.SetBucketPrefix()`, the driver will prepend that prefix to the bucket name.
+
+Here is an example of a session metric and a node metric without a bucket prefix and how their structure looks like:
+
+| Bucket                            | Name                    | Metric                                 |
+|-----------------------------------|-------------------------|----------------------------------------|
+| `s0`                              | `connected-nodes`       | `SessionMetric.Gauges.ConnectedNodes`  |
+| `s0.nodes.127_0_0_1:9042`         | `pool.open-connections` | `NodeMetric.Gauges.OpenConnections`    |
+
+Here are the same examples but with a bucket prefix set in the `Builder`:
+
+| Bucket                            | Name                    | Metric                                 |
+|-----------------------------------|-------------------------|----------------------------------------|
+| `prefix.s0`                       | `connected-nodes`       | `SessionMetric.Gauges.ConnectedNodes`  |
+| `prefix.s0.nodes.127_0_0_1:9042`  | `pool.open-connections` | `NodeMetric.Gauges.OpenConnections`    |
+
+## Configuration
+
+By default, if `Builder.WithMetrics` is not called by the client application, metrics are disabled.
+
+If `Builder.WithMetrics(provider)` is called, the default metrics will be enabled. The default metrics are every metrics except those of type `Timer`. The reasoning behind this is the fact that enabling `Timer` metrics have the potential effect of reducing the throughput of the driver so it's recommended to benchmark the client application with `Timer` metrics enabled before enabling them in production.
```

Comment:
We could use "When" instead of "If":

```suggestion
When `Builder.WithMetrics(provider)` is called, the default metrics will be enabled. The default metrics are composed of every metric except those of type `Timer`. The reasoning behind this is the fact that enabling `Timer` metrics might increase CPU usage and impact throughput of the driver, so it's recommended to benchmark the client application with `Timer` metrics enabled before enabling them in production.
```